app/controllers/api/v1/moves_controller.rb

```diff
@@ -4,10 +4,10 @@ module Api
   module V1
     class MovesController < ApiController
       def index
-        moves_params = Moves::ParamsValidator.new(params[:filter])
+        moves_params = Moves::ParamsValidator.new(filter_params, params[:sort] || {})
```

Comment:
Filters and params used for the model are different entities,
so I think the validation for the `filter` should be apart from the move validation (the model).

The advantage is separation of concern and the chance to reuse the filter validator (high chance)

Reply:
YAGNI